chessboard/olegbet/chessboard.go

```diff
@@ -0,0 +1,54 @@
+package chessboard
+
+// Declare a type named Rank which stores if a square is occupied by a piece - this will be a slice of bools
+
+type Rank []bool
+
+// Declare a type named Chessboard contains a map of eight Ranks, accessed with values from "A" to "H"
+
+type Chessboard map[string]Rank
+
+// CountInRank returns how many squares are occupied in the chessboard,
+// within the given rank
+func CountInRank(cb Chessboard, rank string) (count int) {
+	r := cb[rank]
```

Comment:
I think that there should be a check to see if `rank` is a key in `cb`. It would be something like `r, ok := cb[rank]`. `ok` is boolean of if `rank` is key in `cb`.